### ISSUE TITLE
Callbacks have wrong parameters - mismatch with docs

### ISSUE THREAD

**Issue by mastef** [rvera/image-picker#132]
The documentation mentions that the callbacks are :
```js
changed: function(select, newValues, oldValues, event){...}
clicked: function(select, picker option, event){...}
selected: function(select, picker option, event){...}
```

However the current behaviour is actually : 
```js
changed: function(newValues, oldValues, event){...}
clicked: function(picker option, event){...}
selected: function(picker option, event){...}
```

This is because the callbacks are called with `.call`:
```js
this.opts.changed.call(this.select, old_values, this.selected_values(), original_event);
this.opts.clicked.call(this.picker.select, this, event);
this.opts.changed.call(this.select, old_values, this.selected_values(), original_event);
```

`call()` uses the first parameter as the scope. So actually what happens is that the value of `select` becomes the scope of the function, and not the first parameter.

**Fix 1**
To fix this ( not sure you'd want to fix this, as it might break existing implementations ), you'd want to change it to : 
```js
this.opts.changed.call(this, this.select, old_values, this.selected_values(), original_event);
this.opts.clicked.call(this, this.picker.select, this, event);
this.opts.changed.call(this, this.select, old_values, this.selected_values(), original_event);
```

**Fix 2**
It might be easier to fix the documentation instead, and move `select` to the end of the parameters - this would not break backwards compatibility.
```js
this.opts.changed.call(this.select, old_values, this.selected_values(), original_event, this.select);
this.opts.clicked.call(this.picker.select, this, event, this.picker.select);
this.opts.changed.call(this.select, old_values, this.selected_values(), original_event, this.select);
```
The docs would have to then be changed to :
```js
changed: function(newValues, oldValues, event, select){...}
clicked: function(picker option, event, select){...}
selected: function(picker option, event, select){...}
```

**Fix 3**
Or just change the docs to document the current behaviour, and that the `select` element is available on `this` instead of the first parameter.

**Comment by mastef** [comment]
Additionally the docs are wrong for `changed` as `newValues` and `oldValues` is reversed.
`function(select, newValues, oldValues, event){...}`
should be :
`function(select, oldValues, newValues, event){...} `

**Comment by Humni** [comment]
@mastef Thanks for reporting these issues, I'll get these fixed up before releasing 0.4.0

**Comment by Humni** [comment]
The docs have been updated to reflect current functionality. d86f9f1 resolved this issue and will be merged into the main docs in the 0.4.0 release.
